frontend/src/metabase/public/containers/PublicQuestion/PublicQuestion.jsx

```diff
@@ -166,6 +174,15 @@ class PublicQuestionInner extends Component {
     }
   };
 
+  getCardUiParameters() {
```

Comment:
- I think we could name this `getParameters` to avoid confusion.
- I think this method is needed to encapsulate the need for both `metadata` prop and `card` state.
- Agree with another line expecting this function to return non-array. We could make this function return an empty array instead when there is no `card`

Reply:
I am keeping the method, which is now renamed, and I have changed how we call it. Thanks!